src/components/SubsectionNav/index.tsx

```diff
@@ -77,237 +58,89 @@ const getNavTree = (
-          id={item.data.id}
-          className={clsx("list", !(isRoot || open) && "list-closed")}
-        >
-          <li className="list-item">
-            <WrappedListItem
-              key={item.data.id}
-              text="Overview"
-              label={`Overview of ${item.data.frontmatter.title}`}
-              url={item.data.fields.slug}
-              open={false}
-            />
-          </li>
-          {item.children.map((child: TreeItem, index: number) => (
-            // eslint-disable-next-line react/no-array-index-key
-            <ListChildren item={child} key={index} />
-          ))}
-        </ul>
-      </li>
-    );
-  }
-  return (
-    <li key={item.data.id} className="list-item">
-      <ListItem item={item} onClick={toggleOpen} open={isRoot ? true : open} />
-    </li>
-  );
-};
-
 const SubsectionNav: React.FC<SubsectionNavProps> = ({
   allStructuredNav,
   section,
   currentPage,
 }) => {
   const currentNavSection = getNavTree(allStructuredNav, section, currentPage);
 
+  const isRoot = currentNavSection?.data.fields.navParent === "NONE";
+
   const [responsiveNavOpen, setResponsiveNavOpen] = useState(false);
-  const [hasMounted, setHasMounted] = useState(false);
-  const [navHeight, setNavHeight] = useState(0);
 
   const toggleResponsiveNavOpen = () =>
     setResponsiveNavOpen(!responsiveNavOpen);
 
-  const updateNavigationHeight = () => {
-    const cookieBanner = document.querySelector("#cookie-banner");
-    const cookieBannerHeight = cookieBanner?.clientHeight || 0;
-    const topNav = document.querySelector("#site-top-nav");
-    const topNavHeight = topNav?.clientHeight || 0;
-    const classificationBanner = document.querySelector("#site-banner");
-    const classificationBannerHeight = classificationBanner?.clientHeight || 0;
+  const handleNavigation = (url: string) => {
+    navigate(url);
+  };
+
+  const isBrowser = () => typeof window !== "undefined";
+
+  const isCurrentPage = (url: string, isParentPage: boolean) =>
+    isBrowser() &&
+    (window.location.pathname === url ||
+      window.location.pathname === `${url}/` ||
+      (!isParentPage && window.location.pathname.includes(url)));
+
+  const renderTreeItems = (item: TreeItem) => {
+    let hasChildren = item.children && item.children.length > 0;
 
-    if (window.scrollY > cookieBannerHeight + topNavHeight) {
-      setNavHeight(window.innerHeight - classificationBannerHeight);
-    } else {
-      const size =
-        window.innerHeight -
-        (cookieBannerHeight + topNavHeight - window.scrollY) -
-        classificationBannerHeight;
-      setNavHeight(size);
+    if (item.data.fields.navSection === "components" && hasChildren) {
+      hasChildren = false;
     }
+
+    return (
+      <IcTreeItem
+        key={item.data.id}
+        label={item.data.frontmatter.title}
+        selected={!hasChildren && isCurrentPage(item.data.fields.slug, false)}
+        {...(!hasChildren && {
+          onClick: (e) => {
```

Comment:
I'm struggling to interact with the tree items using the keyboard, is an onKeyPress or something needed here to enable this functionality?